### ISSUE TITLE
Consider using keyed Blake instead of prepend the tag?

### ISSUE THREAD

**Issue by AnomalRoil** [drand/kyber#32]
Currently we prepend a fixed tag to our mesages when we are hashing data:
https://github.com/drand/kyber/blob/a780ab21355ebe7f60b441a586d5e73a40c564eb/encrypt/ibe/ibe.go#L132-L138

I think we should consider using keyed Blake instead of prepend the tag. It would be as simply as replacing the `nil` in the Blake2s instantiation with the `H3Tag`.

@nikkolasg  It would simplify a bit the code and also, keyed hash functions are there exactly for this usecase, no?

**Comment by nikkolasg** [comment]
I'd like to have the possibility to run any hash function and methods we want, extract this bit into a generic or something or config, so we can change it.
Here's blake2s but if we want to use this in a solidity context, I think we might want to go back to like sha256 or keccak and keep the simple "tags" to differentiate the "hash functions" as in the paper.
Maybe something like 
```go
IbeHashing interface {
   H1() hash.Hash
   H2() hash.Hash
   H3() hash.Hash
}
```

**Comment by AnomalRoil** [comment]
Solved in #34 I'd say.
